internal/ansible/handler/logging_enqueue_owner.go

```diff
@@ -1,63 +1,222 @@
-// Copyright 2021 The Operator-SDK Authors
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//	http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+/*
+	Copyright 2018 The Kubernetes Authors.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+			http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
 package handler
 
 import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/util/workqueue"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/event"
-	crHandler "sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-// LoggingEnqueueRequestForOwner wraps operator-lib handler for
-// "InstrumentedEnqueueRequestForObject", and logs the events as they occur
+// Note: The following code is mostly copied from controller-runtime with some additional logging.
+//
+// This can be removed in support of controller-runtime's
+// enqueue request for handler when there is support for
+// wrapping handlers.
+// See this pending PR: https://github.com/kubernetes-sigs/controller-runtime/pull/2427
```

Comment:
I don't think this PR is going to be accepted the way it is. We would need to enable c-r handlers to allow taking wrapped handlers and execute them serially. That's a new feature. I would suggest we open a issue and track it in this repo as a tech debt.

Reply:
I made this issue in this repo: https://github.com/operator-framework/ansible-operator-plugins/issues/40 Do we also need an issue in c-r to track?

Reply:
I guess it should be fine for now. When we have time to pick it up, we can create look into next steps further.